Formula/gobject-introspection-utils.rb

```diff
@@ -0,0 +1,101 @@
+class GobjectIntrospectionUtils < Formula
+  include Language::Python::Shebang
+
+  desc "Generate introspection data for GObject libraries"
+  homepage "https://gi.readthedocs.io/en/latest/"
+  url "https://download.gnome.org/sources/gobject-introspection/1.72/gobject-introspection-1.72.0.tar.xz"
+  sha256 "02fe8e590861d88f83060dd39cda5ccaa60b2da1d21d0f95499301b186beaabc"
+  license all_of: ["GPL-2.0-or-later", "LGPL-2.0-or-later", "MIT"]
+
+  depends_on "bison" => :build
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "cairo"
+  depends_on "glib"
+  depends_on "gobject-introspection"
+  depends_on "libffi"
+  depends_on "pkg-config"
+  depends_on "python@3.10"
+
+  uses_from_macos "flex" => :build
+
+  resource "tutorial" do
+    url "https://gist.github.com/7a0023656ccfe309337a.git",
+        revision: "499ac89f8a9ad17d250e907f74912159ea216416"
+  end
+
+  def install
+    ENV["GI_SCANNER_DISABLE_CACHE"] = "true"
+    inreplace "giscanner/transformer.py", "/usr/share", "#{HOMEBREW_PREFIX}/share"
+    inreplace "meson.build",
+      "config.set_quoted('GOBJECT_INTROSPECTION_LIBDIR', join_paths(get_option('prefix'), get_option('libdir')))",
+      "config.set_quoted('GOBJECT_INTROSPECTION_LIBDIR', '#{HOMEBREW_PREFIX}/lib')"
+
+    # Point the headers and libraries to `gobject-introspection`.
+    # The headers and libraries will be removed later because they are provided by `gobject-introspection`.
+    gobject_introspection = Formula["gobject-introspection"]
+    args = std_meson_args.delete_if do |arg|
+      arg.start_with?("--includedir=")
+    end
+    args += %W[
+      --includedir=#{gobject_introspection.opt_include}
+      --datadir=#{gobject_introspection.opt_pkgshare}
+    ]
+
+    mkdir "build" do
+      system "meson", *args,
+        "-Dpython=#{Formula["python@3.10"].opt_bin}/python3",
+        ".."
+      system "ninja", "-v"
+
+      # Skip files already in gobject-introspection.
+      Formula["meson"].opt_libexec.cd do
+        system "bin/python3", "-c", pyscript
+      end
+      system "ninja", "install", "-v"
```

Comment:
```suggestion
      system "ninja", "install", "-v"
      rewrite_shebang detected_python_shebang, *bin.children
```

Reply:
If I do the shebang rewrite up here instead of after deleting the non-Python files, the Python scripts end up being deleted on Linux and macOS because their shebangs are are too long.  That's how it was also done in the `glib-utils` formula so it makes sense why we have to do it here too.